// @esri/calcite-components-react/dist/components.js
/// <reference types="@esri/calcite-components" preserve="true" />
import React from "react";
import { createComponent } from "@lit/react";
import { makeReactWrapperFactory, getReactWrapperOptions } from "@arcgis/lumina";
const createWrapper = /*@__PURE__*/ makeReactWrapperFactory(React, createComponent);
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-accordion /> directly. */
export const CalciteAccordion = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-accordion", {}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-accordion-item /> directly. */
export const CalciteAccordionItem = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-accordion-item", {
    onCalciteAccordionItemCollapse: "calciteAccordionItemCollapse",
    onCalciteAccordionItemExpand: "calciteAccordionItemExpand",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-action /> directly. */
export const CalciteAction = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-action", {}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-action-bar /> directly. */
export const CalciteActionBar = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-action-bar", {
    onCalciteActionBarCollapse: "calciteActionBarCollapse",
    onCalciteActionBarExpand: "calciteActionBarExpand",
    onCalciteActionBarToggle: "calciteActionBarToggle",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-action-group /> directly. */
export const CalciteActionGroup = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-action-group", {
    onCalciteActionGroupCollapse: "calciteActionGroupCollapse",
    onCalciteActionGroupExpand: "calciteActionGroupExpand",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-action-menu /> directly. */
export const CalciteActionMenu = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-action-menu", {
    onCalciteActionMenuCollapse: "calciteActionMenuCollapse",
    onCalciteActionMenuExpand: "calciteActionMenuExpand",
    onCalciteActionMenuOpen: "calciteActionMenuOpen",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-action-pad /> directly. */
export const CalciteActionPad = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-action-pad", {
    onCalciteActionPadCollapse: "calciteActionPadCollapse",
    onCalciteActionPadExpand: "calciteActionPadExpand",
    onCalciteActionPadToggle: "calciteActionPadToggle",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-alert /> directly. */
export const CalciteAlert = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-alert", {
    onCalciteAlertBeforeClose: "calciteAlertBeforeClose",
    onCalciteAlertBeforeOpen: "calciteAlertBeforeOpen",
    onCalciteAlertClose: "calciteAlertClose",
    onCalciteAlertOpen: "calciteAlertOpen",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-autocomplete /> directly. */
export const CalciteAutocomplete = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-autocomplete", {
    onCalciteAutocompleteBeforeClose: "calciteAutocompleteBeforeClose",
    onCalciteAutocompleteBeforeOpen: "calciteAutocompleteBeforeOpen",
    onCalciteAutocompleteChange: "calciteAutocompleteChange",
    onCalciteAutocompleteClose: "calciteAutocompleteClose",
    onCalciteAutocompleteOpen: "calciteAutocompleteOpen",
    onCalciteAutocompleteTextChange: "calciteAutocompleteTextChange",
    onCalciteAutocompleteTextInput: "calciteAutocompleteTextInput",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-autocomplete-item /> directly. */
export const CalciteAutocompleteItem = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-autocomplete-item", {
    onCalciteAutocompleteItemSelect: "calciteAutocompleteItemSelect",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-autocomplete-item-group /> directly. */
export const CalciteAutocompleteItemGroup = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-autocomplete-item-group", {}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-avatar /> directly. */
export const CalciteAvatar = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-avatar", {}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-block /> directly. */
export const CalciteBlock = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-block", {
    onCalciteBlockBeforeClose: "calciteBlockBeforeClose",
    onCalciteBlockBeforeOpen: "calciteBlockBeforeOpen",
    onCalciteBlockClose: "calciteBlockClose",
    onCalciteBlockCollapse: "calciteBlockCollapse",
    onCalciteBlockExpand: "calciteBlockExpand",
    onCalciteBlockOpen: "calciteBlockOpen",
    onCalciteBlockSortHandleBeforeClose: "calciteBlockSortHandleBeforeClose",
    onCalciteBlockSortHandleBeforeOpen: "calciteBlockSortHandleBeforeOpen",
    onCalciteBlockSortHandleClose: "calciteBlockSortHandleClose",
    onCalciteBlockSortHandleOpen: "calciteBlockSortHandleOpen",
    onCalciteBlockToggle: "calciteBlockToggle",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-block-group /> directly. */
export const CalciteBlockGroup = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-block-group", {
    onCalciteBlockGroupDragEnd: "calciteBlockGroupDragEnd",
    onCalciteBlockGroupDragStart: "calciteBlockGroupDragStart",
    onCalciteBlockGroupMoveHalt: "calciteBlockGroupMoveHalt",
    onCalciteBlockGroupOrderChange: "calciteBlockGroupOrderChange",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-block-section /> directly. */
export const CalciteBlockSection = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-block-section", {
    onCalciteBlockSectionCollapse: "calciteBlockSectionCollapse",
    onCalciteBlockSectionExpand: "calciteBlockSectionExpand",
    onCalciteBlockSectionToggle: "calciteBlockSectionToggle",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-button /> directly. */
export const CalciteButton = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-button", {}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-card /> directly. */
export const CalciteCard = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-card", {
    onCalciteCardSelect: "calciteCardSelect",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-card-group /> directly. */
export const CalciteCardGroup = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-card-group", {
    onCalciteCardGroupSelect: "calciteCardGroupSelect",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-carousel /> directly. */
export const CalciteCarousel = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-carousel", {
    onCalciteCarouselChange: "calciteCarouselChange",
    onCalciteCarouselPause: "calciteCarouselPause",
    onCalciteCarouselPlay: "calciteCarouselPlay",
    onCalciteCarouselResume: "calciteCarouselResume",
    onCalciteCarouselStop: "calciteCarouselStop",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-carousel-item /> directly. */
export const CalciteCarouselItem = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-carousel-item", {}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-checkbox /> directly. */
export const CalciteCheckbox = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-checkbox", {
    onCalciteCheckboxChange: "calciteCheckboxChange",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-chip /> directly. */
export const CalciteChip = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-chip", {
    onCalciteChipClose: "calciteChipClose",
    onCalciteChipSelect: "calciteChipSelect",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-chip-group /> directly. */
export const CalciteChipGroup = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-chip-group", {
    onCalciteChipGroupSelect: "calciteChipGroupSelect",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-color-picker /> directly. */
export const CalciteColorPicker = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-color-picker", {
    onCalciteColorPickerChange: "calciteColorPickerChange",
    onCalciteColorPickerInput: "calciteColorPickerInput",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-color-picker-hex-input /> directly. */
export const CalciteColorPickerHexInput = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-color-picker-hex-input", {
    onCalciteColorPickerHexInputChange: "calciteColorPickerHexInputChange",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-color-picker-swatch /> directly. */
export const CalciteColorPickerSwatch = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-color-picker-swatch", {}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-combobox /> directly. */
export const CalciteCombobox = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-combobox", {
    onCalciteComboboxBeforeClose: "calciteComboboxBeforeClose",
    onCalciteComboboxBeforeOpen: "calciteComboboxBeforeOpen",
    onCalciteComboboxChange: "calciteComboboxChange",
    onCalciteComboboxChipClose: "calciteComboboxChipClose",
    onCalciteComboboxClose: "calciteComboboxClose",
    onCalciteComboboxFilterChange: "calciteComboboxFilterChange",
    onCalciteComboboxOpen: "calciteComboboxOpen",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-combobox-item /> directly. */
export const CalciteComboboxItem = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-combobox-item", {
    onCalciteComboboxItemChange: "calciteComboboxItemChange",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-combobox-item-group /> directly. */
export const CalciteComboboxItemGroup = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-combobox-item-group", {}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-date-picker /> directly. */
export const CalciteDatePicker = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-date-picker", {
    onCalciteDatePickerChange: "calciteDatePickerChange",
    onCalciteDatePickerRangeChange: "calciteDatePickerRangeChange",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-date-picker-day /> directly. */
export const CalciteDatePickerDay = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-date-picker-day", {}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-date-picker-month /> directly. */
export const CalciteDatePickerMonth = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-date-picker-month", {}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-dialog /> directly. */
export const CalciteDialog = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-dialog", {
    onCalciteDialogBeforeClose: "calciteDialogBeforeClose",
    onCalciteDialogBeforeOpen: "calciteDialogBeforeOpen",
    onCalciteDialogClose: "calciteDialogClose",
    onCalciteDialogOpen: "calciteDialogOpen",
    onCalciteDialogScroll: "calciteDialogScroll",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-dropdown /> directly. */
export const CalciteDropdown = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-dropdown", {
    onCalciteDropdownBeforeClose: "calciteDropdownBeforeClose",
    onCalciteDropdownBeforeOpen: "calciteDropdownBeforeOpen",
    onCalciteDropdownClose: "calciteDropdownClose",
    onCalciteDropdownOpen: "calciteDropdownOpen",
    onCalciteDropdownSelect: "calciteDropdownSelect",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-dropdown-group /> directly. */
export const CalciteDropdownGroup = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-dropdown-group", {}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-dropdown-item /> directly. */
export const CalciteDropdownItem = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-dropdown-item", {
    onCalciteDropdownItemSelect: "calciteDropdownItemSelect",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-fab /> directly. */
export const CalciteFab = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-fab", {}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-filter /> directly. */
export const CalciteFilter = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-filter", {
    onCalciteFilterChange: "calciteFilterChange",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-flow /> directly. */
export const CalciteFlow = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-flow", {}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-flow-item /> directly. */
export const CalciteFlowItem = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-flow-item", {
    onCalciteFlowItemBack: "calciteFlowItemBack",
    onCalciteFlowItemClose: "calciteFlowItemClose",
    onCalciteFlowItemCollapse: "calciteFlowItemCollapse",
    onCalciteFlowItemExpand: "calciteFlowItemExpand",
    onCalciteFlowItemScroll: "calciteFlowItemScroll",
    onCalciteFlowItemToggle: "calciteFlowItemToggle",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-graph /> directly. */
export const CalciteGraph = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-graph", {}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-handle /> directly. */
export const CalciteHandle = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-handle", {
    onCalciteHandleChange: "calciteHandleChange",
    onCalciteHandleNudge: "calciteHandleNudge",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-icon /> directly. */
export const CalciteIcon = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-icon", {}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-inline-editable /> directly. */
export const CalciteInlineEditable = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-inline-editable", {
    onCalciteInlineEditableEditCancel: "calciteInlineEditableEditCancel",
    onCalciteInlineEditableEditConfirm: "calciteInlineEditableEditConfirm",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-input /> directly. */
export const CalciteInput = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-input", {
    onCalciteInputChange: "calciteInputChange",
    onCalciteInputInput: "calciteInputInput",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-input-date-picker /> directly. */
export const CalciteInputDatePicker = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-input-date-picker", {
    onCalciteInputDatePickerBeforeClose: "calciteInputDatePickerBeforeClose",
    onCalciteInputDatePickerBeforeOpen: "calciteInputDatePickerBeforeOpen",
    onCalciteInputDatePickerChange: "calciteInputDatePickerChange",
    onCalciteInputDatePickerClose: "calciteInputDatePickerClose",
    onCalciteInputDatePickerOpen: "calciteInputDatePickerOpen",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-input-message /> directly. */
export const CalciteInputMessage = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-input-message", {}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-input-number /> directly. */
export const CalciteInputNumber = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-input-number", {
    onCalciteInputNumberChange: "calciteInputNumberChange",
    onCalciteInputNumberInput: "calciteInputNumberInput",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-input-text /> directly. */
export const CalciteInputText = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-input-text", {
    onCalciteInputTextChange: "calciteInputTextChange",
    onCalciteInputTextInput: "calciteInputTextInput",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-input-time-picker /> directly. */
export const CalciteInputTimePicker = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-input-time-picker", {
    onCalciteInputTimePickerBeforeClose: "calciteInputTimePickerBeforeClose",
    onCalciteInputTimePickerBeforeOpen: "calciteInputTimePickerBeforeOpen",
    onCalciteInputTimePickerChange: "calciteInputTimePickerChange",
    onCalciteInputTimePickerClose: "calciteInputTimePickerClose",
    onCalciteInputTimePickerOpen: "calciteInputTimePickerOpen",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-input-time-zone /> directly. */
export const CalciteInputTimeZone = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-input-time-zone", {
    onCalciteInputTimeZoneBeforeClose: "calciteInputTimeZoneBeforeClose",
    onCalciteInputTimeZoneBeforeOpen: "calciteInputTimeZoneBeforeOpen",
    onCalciteInputTimeZoneChange: "calciteInputTimeZoneChange",
    onCalciteInputTimeZoneClose: "calciteInputTimeZoneClose",
    onCalciteInputTimeZoneOpen: "calciteInputTimeZoneOpen",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-label /> directly. */
export const CalciteLabel = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-label", {}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-link /> directly. */
export const CalciteLink = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-link", {}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-list /> directly. */
export const CalciteList = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-list", {
    onCalciteListChange: "calciteListChange",
    onCalciteListDragEnd: "calciteListDragEnd",
    onCalciteListDragStart: "calciteListDragStart",
    onCalciteListFilter: "calciteListFilter",
    onCalciteListMoveHalt: "calciteListMoveHalt",
    onCalciteListOrderChange: "calciteListOrderChange",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-list-item /> directly. */
export const CalciteListItem = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-list-item", {
    onCalciteListItemClose: "calciteListItemClose",
    onCalciteListItemCollapse: "calciteListItemCollapse",
    onCalciteListItemExpand: "calciteListItemExpand",
    onCalciteListItemSelect: "calciteListItemSelect",
    onCalciteListItemSortHandleBeforeClose: "calciteListItemSortHandleBeforeClose",
    onCalciteListItemSortHandleBeforeOpen: "calciteListItemSortHandleBeforeOpen",
    onCalciteListItemSortHandleClose: "calciteListItemSortHandleClose",
    onCalciteListItemSortHandleOpen: "calciteListItemSortHandleOpen",
    onCalciteListItemToggle: "calciteListItemToggle",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-list-item-group /> directly. */
export const CalciteListItemGroup = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-list-item-group", {}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-loader /> directly. */
export const CalciteLoader = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-loader", {}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-menu /> directly. */
export const CalciteMenu = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-menu", {}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-menu-item /> directly. */
export const CalciteMenuItem = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-menu-item", {
    onCalciteMenuItemSelect: "calciteMenuItemSelect",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-meter /> directly. */
export const CalciteMeter = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-meter", {}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-navigation /> directly. */
export const CalciteNavigation = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-navigation", {
    onCalciteNavigationActionSelect: "calciteNavigationActionSelect",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-navigation-logo /> directly. */
export const CalciteNavigationLogo = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-navigation-logo", {}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-navigation-user /> directly. */
export const CalciteNavigationUser = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-navigation-user", {}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-notice /> directly. */
export const CalciteNotice = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-notice", {
    onCalciteNoticeBeforeClose: "calciteNoticeBeforeClose",
    onCalciteNoticeBeforeOpen: "calciteNoticeBeforeOpen",
    onCalciteNoticeClose: "calciteNoticeClose",
    onCalciteNoticeOpen: "calciteNoticeOpen",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-option /> directly. */
export const CalciteOption = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-option", {}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-option-group /> directly. */
export const CalciteOptionGroup = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-option-group", {}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-pagination /> directly. */
export const CalcitePagination = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-pagination", {
    onCalcitePaginationChange: "calcitePaginationChange",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-panel /> directly. */
export const CalcitePanel = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-panel", {
    onCalcitePanelClose: "calcitePanelClose",
    onCalcitePanelCollapse: "calcitePanelCollapse",
    onCalcitePanelExpand: "calcitePanelExpand",
    onCalcitePanelScroll: "calcitePanelScroll",
    onCalcitePanelToggle: "calcitePanelToggle",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-popover /> directly. */
export const CalcitePopover = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-popover", {
    onCalcitePopoverBeforeClose: "calcitePopoverBeforeClose",
    onCalcitePopoverBeforeOpen: "calcitePopoverBeforeOpen",
    onCalcitePopoverClose: "calcitePopoverClose",
    onCalcitePopoverOpen: "calcitePopoverOpen",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-progress /> directly. */
export const CalciteProgress = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-progress", {}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-radio-button /> directly. */
export const CalciteRadioButton = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-radio-button", {
    onCalciteRadioButtonChange: "calciteRadioButtonChange",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-radio-button-group /> directly. */
export const CalciteRadioButtonGroup = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-radio-button-group", {
    onCalciteRadioButtonGroupChange: "calciteRadioButtonGroupChange",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-rating /> directly. */
export const CalciteRating = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-rating", {
    onCalciteRatingChange: "calciteRatingChange",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-scrim /> directly. */
export const CalciteScrim = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-scrim", {}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-segmented-control /> directly. */
export const CalciteSegmentedControl = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-segmented-control", {
    onCalciteSegmentedControlChange: "calciteSegmentedControlChange",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-segmented-control-item /> directly. */
export const CalciteSegmentedControlItem = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-segmented-control-item", {}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-select /> directly. */
export const CalciteSelect = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-select", {
    onCalciteSelectChange: "calciteSelectChange",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-sheet /> directly. */
export const CalciteSheet = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-sheet", {
    onCalciteSheetBeforeClose: "calciteSheetBeforeClose",
    onCalciteSheetBeforeOpen: "calciteSheetBeforeOpen",
    onCalciteSheetClose: "calciteSheetClose",
    onCalciteSheetOpen: "calciteSheetOpen",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-shell /> directly. */
export const CalciteShell = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-shell", {}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-shell-panel /> directly. */
export const CalciteShellPanel = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-shell-panel", {
    onCalciteShellPanelCollapse: "calciteShellPanelCollapse",
    onCalciteShellPanelExpand: "calciteShellPanelExpand",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-slider /> directly. */
export const CalciteSlider = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-slider", {
    onCalciteSliderChange: "calciteSliderChange",
    onCalciteSliderInput: "calciteSliderInput",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-sort-handle /> directly. */
export const CalciteSortHandle = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-sort-handle", {
    onCalciteSortHandleAdd: "calciteSortHandleAdd",
    onCalciteSortHandleBeforeClose: "calciteSortHandleBeforeClose",
    onCalciteSortHandleBeforeOpen: "calciteSortHandleBeforeOpen",
    onCalciteSortHandleClose: "calciteSortHandleClose",
    onCalciteSortHandleMove: "calciteSortHandleMove",
    onCalciteSortHandleOpen: "calciteSortHandleOpen",
    onCalciteSortHandleReorder: "calciteSortHandleReorder",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-sortable-list /> directly. */
export const CalciteSortableList = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-sortable-list", {
    onCalciteListOrderChange: "calciteListOrderChange",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-split-button /> directly. */
export const CalciteSplitButton = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-split-button", {
    onCalciteSplitButtonPrimaryClick: "calciteSplitButtonPrimaryClick",
    onCalciteSplitButtonSecondaryClick: "calciteSplitButtonSecondaryClick",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-stack /> directly. */
export const CalciteStack = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-stack", {}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-stepper /> directly. */
export const CalciteStepper = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-stepper", {
    onCalciteStepperChange: "calciteStepperChange",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-stepper-item /> directly. */
export const CalciteStepperItem = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-stepper-item", {
    onCalciteStepperItemSelect: "calciteStepperItemSelect",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-swatch /> directly. */
export const CalciteSwatch = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-swatch", {
    onCalciteSwatchSelect: "calciteSwatchSelect",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-swatch-group /> directly. */
export const CalciteSwatchGroup = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-swatch-group", {
    onCalciteSwatchGroupSelect: "calciteSwatchGroupSelect",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-switch /> directly. */
export const CalciteSwitch = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-switch", {
    onCalciteSwitchChange: "calciteSwitchChange",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-tab /> directly. */
export const CalciteTab = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-tab", {}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-tab-nav /> directly. */
export const CalciteTabNav = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-tab-nav", {
    onCalciteTabChange: "calciteTabChange",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-tab-title /> directly. */
export const CalciteTabTitle = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-tab-title", {
    onCalciteTabsActivate: "calciteTabsActivate",
    onCalciteTabsClose: "calciteTabsClose",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-table /> directly. */
export const CalciteTable = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-table", {
    onCalciteTablePageChange: "calciteTablePageChange",
    onCalciteTableSelect: "calciteTableSelect",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-table-cell /> directly. */
export const CalciteTableCell = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-table-cell", {}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-table-header /> directly. */
export const CalciteTableHeader = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-table-header", {}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-table-row /> directly. */
export const CalciteTableRow = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-table-row", {
    onCalciteTableRowSelect: "calciteTableRowSelect",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-tabs /> directly. */
export const CalciteTabs = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-tabs", {}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-text-area /> directly. */
export const CalciteTextArea = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-text-area", {
    onCalciteTextAreaChange: "calciteTextAreaChange",
    onCalciteTextAreaInput: "calciteTextAreaInput",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-tile /> directly. */
export const CalciteTile = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-tile", {
    onCalciteTileSelect: "calciteTileSelect",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-tile-group /> directly. */
export const CalciteTileGroup = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-tile-group", {
    onCalciteTileGroupSelect: "calciteTileGroupSelect",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-time-picker /> directly. */
export const CalciteTimePicker = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-time-picker", {
    onCalciteTimePickerChange: "calciteTimePickerChange",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-tooltip /> directly. */
export const CalciteTooltip = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-tooltip", {
    onCalciteTooltipBeforeClose: "calciteTooltipBeforeClose",
    onCalciteTooltipBeforeOpen: "calciteTooltipBeforeOpen",
    onCalciteTooltipClose: "calciteTooltipClose",
    onCalciteTooltipOpen: "calciteTooltipOpen",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-tree /> directly. */
export const CalciteTree = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-tree", {
    onCalciteTreeSelect: "calciteTreeSelect",
}));
/** @deprecated since 5.0. React wrappers are not needed in React 19 or above. Use <calcite-tree-item /> directly. */
export const CalciteTreeItem = /*@__PURE__*/ createWrapper(getReactWrapperOptions("calcite-tree-item", {
    onCalciteTreeItemCollapse: "calciteTreeItemCollapse",
    onCalciteTreeItemExpand: "calciteTreeItemExpand",
    onCalciteTreeItemSelect: "calciteTreeItemSelect",
}));
